### ISSUE TITLE
https://noblemtl.com/

### ISSUE THREAD

**Issue by Iadders** [dipu-bd/lightnovel-crawler#1539]
Yes

 Yes

- Language: English
- Supports Searching: Yes
- Contains Machine Translations: Yes
- Contains Manga/Manhua/Manhwa: No
- Has CloudFlare Protection: Yes

**Comment by Iadders** [comment]
#1540

Is this source rejected?

**Comment by idMysteries** [comment]
Maybe dipu-bd can fix the problem. It doesn't seem to work for me because of the protection of cloudfrale

**Comment by Iadders** [comment]
Can you use the Parser from webtoepub as a reference? 

`"use strict";

parserFactory.register("noblemtl.com", () => new NoblemtlParser());
parserFactory.register("tamagotl.com", () => new NoblemtlParser());

class NoblemtlParser extends Parser{
    constructor() {
        super();
    }

    async getChapterUrls(dom) {
        return [...dom.querySelectorAll("div.eplister a")]
            .map(this.linkToChapter)
            .reverse()
    }

    linkToChapter(link) {
        let title = link.querySelector(".epl-num").textContent + " "
            + link.querySelector(".epl-title").textContent;
        return ({
            sourceUrl:  link.href,
            title: title
        });
    }

    findContent(dom) {
        return dom.querySelector(".entry-content");
    }

    extractTitleImpl(dom) {
        return dom.querySelector("h1.entry-title");
    }

    findChapterTitle(dom) {
        return dom.querySelector("h1.entry-title");
    }

    findCoverImageUrl(dom) {
        return util.getFirstImgSrc(dom, ".thumbook");
    }

    preprocessRawDom(webPageDom) {
        util.removeChildElementsMatchingCss(webPageDom, "div.saboxplugin-wrap");
    }

    getInformationEpubItemChildNodes(dom) {
        return [...dom.querySelectorAll("div.entry-content")];
    }
}
`

**Comment by Iadders** [comment]
Can you use the Parser from webtoepub as reference?

**Comment by Iadders** [comment]
@idMysteries

Can you use the Parser from webtoepub as reference?

https://github.com/dteviot/WebToEpub/issues/803

**Comment by idMysteries** [comment]
No, WebToEpub is extension for Firefox and Chrome. It doesn't need to bypass CloudFlare protection

**Comment by Iadders** [comment]
Lemme know if you get it working.
